common/lib/dependabot/dependency.rb

```diff
@@ -69,7 +74,8 @@ def to_h
         "previous_version" => previous_version,
         "previous_requirements" => previous_requirements,
         "package_manager" => package_manager,
-        "subdependency_metadata" => subdependency_metadata
+        "subdependency_metadata" => subdependency_metadata,
+        "removed" => removed? ? true : nil
```

Comment:
Along the same lines, if we were to have `#removed?` return a boolean, we could just use it here and `"removed"` would always evaluate to `true`/`false`, which feels right _to me_. If there's an angle I'm not considering here, please let me know 😄

Reply:
When false I'm setting it to `nil` here so that the call to `#compact` will remove it from the result. That's to avoid altering any behavior until we've made needed adjustments in other systems and turn the feature on.

Reply:
👍 My next thought would be to make sure we're handling any use of `#fetch` to make sure we provide a fallback value in case it's been `#compact`ed away. Glanced through the diff here and I think we're handling the one case that exists 🙂